src/Akka.Persistence.Sql/Query/QueryThrottler.cs

```diff
@@ -95,19 +173,29 @@ private void QueryStartGranted(IActorRef actorRef)
         actorRef.Tell(Query.QueryStartGranted.Instance);
     }
     
-    private void ReturnQueryPermit(IActorRef actorRef)
+    private void ReturnQueryPermit()
     {
         _usedPermits--;
-        Context.Unwatch(actorRef);
 
         if (_usedPermits < 0)
-            throw new IllegalStateException("Permits must not be negative");
+        {
+            _log.Warning("Permits must not be negative");
```

Comment:
Is it overkill to include the negative value here? Also what justifies the reset to zero?

Reply:
Its a warning for the developers. It should not kill the actor outright, and the throttler should keep on going. If the value goes negative, this means that we over-return permits, something is returning more permits than what was issued. Its not a critical error, the actor should work regardless, but we should take a note of it happening.

Reply:
Makes sense; Should we add a comment in code AFA your comment? Mostly asking in case other maintainers/devs ask same question in future...

Reply:
good idea